### ISSUE TITLE
Sorting State for Clients in UI is incorrect

### ISSUE THREAD

**Issue by omada-jacob-smith** [hashicorp/nomad#6262]
### Nomad version
Nomad v0.9.5+ent (5926e0dc9fec0c13e6fc6f622cea7546ce42a27f)

### Operating system and Environment details
Ubuntu 18.04

### Issue
Sorting of client's state (status) in UI is broken when mixed ineligible/ready clients. 

### Reproduction steps
1. Mark half of the clients ineligible
2. Drain clients from 1
3. After that has been finished, sorting of state will actually be by ID rather than state (status)

<img width="125" alt="Screen Shot 2019-09-04 at 10 27 52 AM" src="https://user-images.githubusercontent.com/46694346/64277286-d3af3200-cefe-11e9-8ef5-675497617972.png">

**Comment by shantanugadgil** [comment]
message

**Comment by backspace** [comment]
Thanks for letting us know about this and I’m sorry for how long it took to respond. I found two causes for this:

1. The changes in #5789 collapsed the status/draining/eligible columns into one but didn’t include anything about sorting. The column is called “state” but is still sorting on “status”.
    * (The nomenclature here is its own confusing thing; there’s also a [`compositeStatus`](https://github.com/hashicorp/nomad/blob/74190ff809afe4ed07f2a0aa71dd58e6161e6453/ui/app/models/node.js#L64-L66) that derives from those properties excluding `isDraining`.)

2. The [`Sortable` mixin](https://github.com/hashicorp/nomad/blob/f6232f181729915a08b8e57325080792462dc266/ui/app/mixins/sortable.js) doesn’t declare the sort properties as dependent keys. Even when I made the sorting based on a new `state` property, changing the properties didn’t change the sort because of this.

I should be able to put together a PR to fix this tomorrow or next week. 💞
